src/PushNotifications/PushNotifications/PushNotificationAppService.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Elders.Cronus.DomainModeling;
+using PushNotifications.Contracts;
+using PushNotifications.Contracts.PushNotifications.Commands;
+
+namespace PushNotifications
+{
+    public class PushNotificationAppService : AggregateRootApplicationService<PushNotification>,
+         ICommandHandler<SendPushNotification>
+    {
+        public void Handle(SendPushNotification command)
+        {
+            // Ignore pushnotifications that expired
+            if (Timestamp.UtcNow().DateTime > command.ExpiresAt.DateTime)
```

Comment:
aren't those commands invalid?
this looks like business logic and it should be in the PushNotification constructor, otherwise it can not be unit tested

Reply:
moved